app/routes/events.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from 'react';
+import { initialEventsList } from '~/components/Calendar';
+import Drawer from '~/components/common/drawer';
+import { CalEvent } from '~/utils/interfaces';
+
+const Events = () => {
+  const [selectedEvent, setSelectedEvent] = useState<CalEvent | null>(null);
+  const [isDrawerVisible, setIsDrawerVisible] = useState(false);
+
+  const toggleDrawer = (event?: CalEvent) => {
+    setSelectedEvent(event || null);
+    setIsDrawerVisible(!isDrawerVisible);
```

Comment:
Nitpick: 
A better approach for this would be to pass an updater function `setIsDrawerVisible(prev => !prev)`  to maintain consistency and prevent any potential bugs if we add multiple updates in this toggleDrawer for some reason. 

Link to [react docs](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state)

Reply:
Yes, thank you. Fixed it.